base/gcutils.jl

```diff
@@ -272,4 +272,17 @@ function logging_enabled()
     ccall(:jl_is_gc_logging_enabled, Cint, ()) != 0
 end
 
+"""
+    GC.generational_mode(on::Bool)
+
+Enable or disable generational collections and object age promotion
+"""
+function generational_mode(on::Bool=true)
+    if on == false
+        gc() #Make sure all live object are old
```

Comment:
I don't think it makes sense to do this on the Julia side. There could be other threads generating garbage. Needs to be done on the C side, possibly holding other threads in safe point.